src/ameba/ast/visitors/scope_visitor.cr

```diff
@@ -3,6 +3,21 @@ require "./base_visitor"
 module Ameba::AST
   # AST Visitor that traverses the source and constructs scopes.
   class ScopeVisitor < BaseVisitor
+    SUPER = "super"
+    # List of special top level macros that use assignments
+    # as the macro arguments thus such arguments are not treated as assignments.
+    #
+    # For example:
+    #
+    # ```
+    # bar = 1
```

Comment:
This line is uneccessary for a demonstration. Ditto two below.

Reply:
`record Foo, bar = 2 # this is not an assignment` says it all.

Reply:
Thanks. I would prefer to leave it as is.